crates/matrix-sdk/tests/integration/room/beacon/mod.rs

```diff
@@ -374,3 +374,95 @@ async fn test_observe_single_live_location_share() {
     assert_eq!(beacon_info.timeout, Duration::from_millis(3000));
     assert_eq!(beacon_info.ts, current_time);
 }
+
+#[async_test]
+async fn test_observing_live_location_does_not_return_own_beacon_updates() {
+    let (client, server) = logged_in_client_with_server().await;
+
+    let mut sync_builder = SyncResponseBuilder::new();
+
+    let current_time = MilliSecondsSinceUnixEpoch::now();
+    let millis_time = current_time
+        .to_system_time()
+        .unwrap()
+        .duration_since(UNIX_EPOCH)
+        .expect("Time went backwards")
+        .as_millis() as u64;
+
+    mock_sync(
```

Comment:
Could this please be rewritten using the `MatrixMockServer`, I now added plenty of docs to it so it shouldn't be hard to figure out how that can be translated to it.

Though if you feel lost don't fret to shout.